cassie/modules/empire.py

```diff
@@ -0,0 +1,340 @@
+import datetime
+
+from urllib.parse import urlparse
+
+from cassie.argparselite import ArgumentParserLite
+from cassie.imcontent import IMContentMarkdown
+from cassie.templates import CassieXMPPBotModule
+
+try:
+	import requests
+except ImportError:
+	print('Failed to import "requests" module.')
+
+class EmpireAPI(object):
+	"""Class to provide access to various functionality exposed through the Empire API"""
+
+	def __init__(self, empire_config):
+		"""Initialize a new instance of the EmpireAPI class"""
+		self.username = empire_config['user']
+		self.password = empire_config['pass']
+		self.api_base_url = empire_config['url'] + 'api/'
+		self.json_request_headers = {'Content-Type':'application/json'}
+		self.verify_server_cert = False
+		self._token = self._get_api_token()
+
+	def _get_api_token(self):
+	    """Get the token for the Empire server"""
+	    server_url = self.api_base_url + 'admin/login'
+	    creds = {"username": self.username, "password": self.password}
+	    response = self.send_post_request(server_url, json=creds)
+	    token = response['token']
+	    return token
+
+	def send_get_request(self, url):
+		"""Send a GET request to the Empire server API"""
+		try:
+			response = requests.get(url, headers=self.json_request_headers, verify=self.verify_server_cert)
+		except requests.exceptions.SSLError:
+			return 'Server certificate validation failed.'
```

Comment:
This seems like a bit of a bad idea, because in the event of an error the return type will be a string vs some kind of other python object if there is not an error. It might be better to handle this exception a bit higher in the call stack and print this message to the user.